### PR TITLE
adding a conditional welcome prompt and imported navigation

### LINKED ISSUE #7: 7. As a user, I want to see a welcoming prompt to add my first item if my list is empty to help me get oriented to how the app works.
## Summary

When a user starts a new list, they need a little guidance on how to get started. Think through what you'd want to see as a user and make some intentional user experience decisions that will guide our users in confidently adding their very first item. 

## Acceptance criteria
- [ ] The list view, when there are no items to display, should show a prompt (e.g., a button) for the user to add their first item

## Notes
- [The wireframe](https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-empty.png) is a little short on the "welcoming" part of the story, but otherwise is functionally complete for the purposes of the AC.

### PR DESCRIPTION
**Pull request by DwightTheShark**
## Description
When a user starts a new list, they need a little guidance on how to get started.  If the list is empty, the user is encouraged to add an item on the Add Item Page.  If they're returning, the user is welcomed back to their list.

## Related Issue

Closes #7 .

## Acceptance Criteria
- [x] The list view, when there are no items to display, should show a prompt (e.g., a button) for the user to add their first item

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Before

Before changes

### After

After changes when list is full
After changes when list is empty
GIF of new feature in action

## Testing Steps / QA Criteria
Try clearing your local memory, and then creating a new list to check the conditional rendering.

## Optimizations
In the future, the placeholder that says "Welcome Back" could instead hold the title of a list, if we allow multiple lists per user.

### CODE REVIEW

**Review by ehammes — APPROVED** [review]
Approved! 
Noticed that Home page doesn't have the latest code update (& firebase updates) from last week. Just pointing out for visibility.

**Review by tinazarb — APPROVED** [review]
There seems to be a branch conflict in src/views/List.jsx that needs to be resolved. Otherwise LGTM

**Comment by ehammes** [comment]
Also there's a console log error for the following: 

> Warning: Invalid DOM property `inputmode`. Did you mean `inputMode`?

@louieknolle you noticed a similar error in last week's PR

**Comment by DwightTheShark** [comment]
Both of these are because I didn't have the latest version to work on and we just ran with it.  We'll get it tidied up :)

File: src/views/List.jsx [hunk 1]
```diff
@@ -17,38 +18,59 @@ export function List({ data }) {
...
+			) : (
+				// otherwise show people their list
+				<div>
+					<h1>Welcome back!</h1>
+					<form>
+						<label htmlFor="filter">Filter Items:</label>
+						<input
+							id="filter"
+							type="text"
+							inputmode="search"
```
Comment (timothy-taylor) [inline src/views/List.jsx]: We should probably push a fix for this before we do the merge conflict on Sunday.

File: src/views/List.jsx [hunk 2]
```diff
@@ -17,38 +18,59 @@ export function List({ data }) {
...
-			</ul>
+			{/* welcome people to add to their list if it's empty */}
+			{data.length === 0 ? (
+				<div>
+					<h1>Hello!</h1>
+					<h2>
+						You have nothing on your list yet! Click here to add your first
+						item:
+					</h2>
+					<button onClick={handleNavigate} type="button">
```
Comment (timothy-taylor) [inline src/views/List.jsx]: This button should probably be replaced by an actual link for accessibility purposes. A button is generally meant to perform an action within a page, whereas the link is sending you to another page. Check out `<NavLink />` from `react-router-dom`.

**Review by timothy-taylor — APPROVED** [review]
Functionality looks great! I left a couple of recommended fixes to look at before Sunday.

**Pull request merged**